### PR TITLE
Fix internal deprecation notices

### PR DESCRIPTION
**Pull request by greg0ire**
I am targetting this branch, because this is BC.

## Changelog

```markdown
### Fixed
- deprecation notices that could not be avoided in the `CoreController` class
```

## Subject

There are still deprecation notices, but they come from Symfony. This PR fixes all outstanding internal deprecation notices.

### CODE REVIEW

File: Controller/CoreController.php [hunk 1]
```diff
@@ -158,10 +155,25 @@ protected function getSearchHandler()
...
     }
+
+    /**
+     * Get the request object from the container.
+     *
+     * @return Request
+     */
+    private function getCurrentRequest()
+    {
+        // NEXT_MAJOR: simplify this when dropping sf < 2.4
```
Comment (core23) [inline Controller/CoreController.php]: What about injecting this in the methods?

Comment (greg0ire) [inline Controller/CoreController.php]: Someone might be using the methods from an action that does not have the request. They will have to inject the request too. Plus, it's a signature change, which means more deprecations.

Comment (core23) [inline Controller/CoreController.php]: Why not reimplement `getRequest` instead of `getCurrentRequest`?

Comment (greg0ire) [inline Controller/CoreController.php]: Because otherwise users will still get the deprecation notices when calling it.

Comment (core23) [inline Controller/CoreController.php]: Don't think so: If we completly overwrite this method, the `trigger_error` won't get called.
https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L318

Comment (greg0ire) [inline Controller/CoreController.php]: What do you mean by "overwriting"?

Comment (greg0ire) [inline Controller/CoreController.php]: If we do as in your example, we will get the deprecation anyway when calling the ourselves from other methods.

Comment (core23) [inline Controller/CoreController.php]: > What do you mean by "overwriting"?

We create a method named `getRequest` and implement the same logic like in your `getCurrentRequest` without calling any `parent::getRequest`. This won't trigger the original (deprecated) method.

The only problem is the visibility, because `getCurrentRequest` is private and `getRequest` is public.

Comment (jordisala1991) [inline Controller/CoreController.php]: If you declare it private no one extending this class will be able to use it. Might be better to change method signature on a new major instead of refactoring that method

Comment (greg0ire) [inline Controller/CoreController.php]: Guys, I no longer understand.

@core23 : I can't create a method named `getRequest`, because it already exists (and contains a deprecation). I can't stop calling `parent::getRequest` because I never call it.

`getCurrentRequest` is private on purpose. I want to have a convenience method for use in other methods of the class. It's not intended for usage by child classes, which should use the `$request` parameter.

@jordisala1991 we don't want people to get the request from the parent class anymore.

Comment (jordisala1991) [inline Controller/CoreController.php]: We are already redefining the getRequest from the Symfony Controller, so we need a new method to retrieve the request without getting a deprecation message.

IMO we could mark it as in SonataMedia: https://github.com/sonata-project/SonataMediaBundle/pull/1189/commits/64426a90eac5aa2b667fc57c912928872a1b3d26. Just to make sure we move to Request injected as a method parameter. What do you think @greg0ire ?

Comment (greg0ire) [inline Controller/CoreController.php]: `getCurrentRequest` is used in a method that is not an action : `getBaseTemplate`. Not sure it's great DX to have people change their action to add a parameter, for the sole purpose of passing it to this method.

Comment (jordisala1991) [inline Controller/CoreController.php]: Agree, this is not really related. Just asking for science. Thanks for explaining! :)

**Comment by greg0ire** [comment]
@sonata-project/contributors please merge this.

**Comment by OskarStark** [comment]
Thank you @greg0ire 👍
